stdpopsim/citations.py

```diff
@@ -23,10 +36,37 @@ class Citation(object):
     doi = attr.ib(type=str, kw_only=True)
     author = attr.ib(type=str, kw_only=True)
     year = attr.ib(type=int, kw_only=True)
+    reasons = attr.ib(factory=set, kw_only=True)
 
     def __str__(self):
         return f"{self.author}, {self.year}: {self.doi}"
 
+    def print(self, file=sys.stdout):
+        print("[", end="", file=file)
+        for i, reason in enumerate(self.reasons):
+            end = "" if i == len(self.reasons)-1 else ", "
+            print(reason, end=end, file=file)
+        print("]", file=file)
+        print("  "+str(self), file=file)
+
+    def because(self, reasons):
```

Comment:
I'm not convinced we need the `because` function. I see that it is taking a logical 'or' of `self.reasons` and `reasons` --- will there be cases where we need to add new reasons to a citation once that citation has been defined?

Reply:
Why yes there is, I'm glad you asked. :)
Its mostly when using DroMel right now, because `Li et al. 2016` can be used for three different reasons, and there are (at least) two reasonable combinations of two of these reasons.

E.g. `stdpopsim DroMel -l0.001 -o /dev/null 10` gives
```
[generation time, population size]
  Li et al., 2006: https://doi.org/10.1371/journal.pgen.0020166
```
While `stdpopsim DroMel -l0.001 -m OutOfAfrica_2L06 -o /dev/null 10` gives
```
[generation time, demographic model]
  Li et al., 2006: https://doi.org/10.1371/journal.pgen.0020166
```

Reply:
ah very nice! I didn't look there, and this offers great flexibility then. I like it.